### PR TITLE
fix: add `predicateData` to predicate resources and inputs

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close: https://github.com/FuelLabs/fuels-ts/issues/2379

### CODE REVIEW

**Review by maschad — APPROVED** [review]
Very nice

File: packages/fuel-gauge/src/predicate/predicate-estimations.test.ts [hunk 1]
```diff
@@ -204,5 +209,50 @@ describe('Predicate', () => {
...
+        transactionRequest.addResources(resources);
+        const inputs = transactionRequest.inputs.filter(isRequestInputResource);
+
+        expect(inputs.length).toBeGreaterThan(0);
+        inputs.forEach((resource) => {
+          expect(resource.predicateData).toBeDefined();
+        });
+      });
+
+      test('fetching with provider', async () => {
```
Comment (nedsalk) [inline packages/fuel-gauge/src/predicate/predicate-estimations.test.ts:231]: Based on the `describe` above titled `should ensure resources inputs from a predicate includes predicateData`, this test `fetching with provider` implies that fetching predicate resources with `provider.getResourcesToSpend` populates the `predicateData`, which doesn't match the contents of the test.

I suggest renaming the `describe` and `test`s, e.g.
- describe: `predicate resource fetching and predicateData population`,
    - t1: `getting predicate resources via the predicate automatically populates predicateData`
    - t2: `getting predicate resources via the provider requires manual predicateData population`

**Review by nedsalk — CHANGES_REQUESTED** [review]
I thought I requested changes but I didn't. The change request is tied to my comment above.

File: packages/fuel-gauge/src/predicate/predicate-estimations.test.ts (hunk 1 above)
Comment (Torres-ssf) [reply packages/fuel-gauge/src/predicate/predicate-estimations.test.ts]: https://github.com/FuelLabs/fuels-ts/pull/2380/commits/4179a9cada578e04ba165b60adda817dd5afa659

**Review by arboleya — DISMISSED** [review]

**Review by nedsalk — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**